### PR TITLE
feat: Add Temporal to the dev and hobby stacks

### PR DESCRIPTION
**Pull request by fuziontech**
## Problem

Add Temporal scheduler to dev and hobby stacks in preparation for workflow development

## Changes

- Adds temporal scheduler to the docker-compose stack
- Adds django management commands
  - For starting a worker
  - For starting a workflow
- Adds temporal worker to the docker-compose hobby stack
- Adds temporal worker to the dev workflow

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## TODO

Still need to fix `plugins` container to include pnpm but that can be in a later PR - this needs to be landed

### CODE REVIEW

**Comment by fuziontech** [comment]
image
!!

File: posthog/temporal/utils.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            heartbeat_task = asyncio.create_task(heartbeat_every(heartbeat_timeout.total_seconds() / 2))
+        try:
+            return await fn(*args, **kwargs)
+        finally:
+            if heartbeat_task:
+                heartbeat_task.cancel()
+                # Wait for heartbeat cancellation to complete
+                await asyncio.wait([heartbeat_task])
+
+    return cast(F, wrapper)
```
Comment (tomasfarias) [inline posthog/temporal/utils.py:28]: A bit nitpicky but `cast` doesn't do anything: 

> This returns the value unchanged. To the type checker this signals that the return value has the designated type, but at runtime we intentionally don’t check anything

I'd prefer if we used the appropriate type hints instead as `cast` could be hiding something.

File: posthog/temporal/runner.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+from posthog.temporal.workflows import NoOpWorkflow
+
+
+async def execute_noop_workflow(host, port):
+    client = await Client.connect(f"{host}:{port}")
+    result = await client.execute_workflow(
+        NoOpWorkflow.run,
+        start_to_close_timeout=timedelta(seconds=60),
+        schedule_to_close_timeout=timedelta(minutes=5),
+    )
```
Comment (tomasfarias) [inline posthog/temporal/runner.py:14]: I believe this call is missing the `time` argument for `NoOpWorkflow.run`.

Comment (fuziontech) [reply posthog/temporal/runner.py]: wooops!

**Pull request merged**